storage/innobase/lock/lock0lock.cc

```diff
@@ -691,31 +776,10 @@ lock_rec_has_to_wait(
 #endif /* HAVE_REPLICATION */
 
 #ifdef WITH_WSREP
-	/* New lock request from a transaction is using unique key
-	scan and this transaction is a wsrep high priority transaction
-	(brute force). If conflicting transaction is also wsrep high
-	priority transaction we should avoid lock conflict because
-	ordering of these transactions is already decided and
-	conflicting transaction will be later replayed. */
-	if (trx->is_wsrep_UK_scan()
-	    && wsrep_thd_is_BF(lock2->trx->mysql_thd, false)) {
-		return false;
-	}
-
-	/* if BF-BF conflict, we have to look at write set order */
-	if (trx->is_wsrep() &&
-	   (type_mode & LOCK_MODE_MASK) == LOCK_X &&
-	   (lock2->type_mode & LOCK_MODE_MASK) == LOCK_X &&
-	   wsrep_thd_order_before(trx->mysql_thd,
-				  lock2->trx->mysql_thd)) {
-		return false;
+	if (trx->is_wsrep()) {
+	  /* We have some additional rules for wsrep transactions. */
+	  return (lock_rec_has_to_wait_wsrep(trx, type_mode, lock2));
 	}
```

Comment:
This function is formatted using the archaic InnoDB formatting rules. Hence we should use TAB instead of spaces for indentation. The comment does not add much value; this should be obvious from the name of the tail-called function. The parentheses after `return` are redundant and should better be omitted.

Reply:
Fixed.